libdns.go

```diff
@@ -105,26 +147,87 @@ type ZoneLister interface {
 }
 
 // Record is a generalized representation of a DNS record.
-//
-// The values of this struct should be free of zone-file-specific syntax,
-// except if this struct's fields do not sufficiently represent all the
-// fields of a certain record type; in that case, the remaining data for
-// which there are not specific fields should be stored in the Value as
-// it appears in the zone file.
 type Record struct {
 	// provider-specific metadata
 	ID string
 
 	// general record fields
-	Type  string
-	Name  string // partially-qualified (relative to zone)
+
+	// The `Type` field specifies the type of the record. Implementations may
+	// or may not support any given record type, and may support additional
+	// "private" record types with implementation-defined behavior.
+	//
+	// Examples: "A", "AAAA", "CNAME", "MX", "TXT"
+	Type string
+
+	// The `Name` field specifies the name of the record. It is partially
+	// qualified relative to the current zone. This field is called a "Label" by
+	// RFC7719. You may use "@" or the empty string "" to represent the root of
+	// the zone.
```

Comment:
> You may use "@" or the empty string "" to represent the root of the zone.

I'm not actually sure how you are currently expected to specify the root of the current zone, so this is just my best guess at what should work.

Reply:
I am not sure either! Is it more conventional to use "" or "@"? Cloudflare, for instance, requires "@", but I don't know if that's a standard or just preference.